### PR TITLE
[DOCS] Document Tree & TreeItem

### PR DESCRIPTION
**Pull request by mhilbrunner**

### CODE REVIEW

File: doc/classes/TreeItem.xml [hunk 1]
```diff
@@ -499,6 +543,7 @@
 			<argument index="1" name="selectable" type="bool">
 			</argument>
 			<description>
+				Sets whether the column [code]column[/code] is selectable.
```
Comment (cbscribe) [inline doc/classes/TreeItem.xml]: Per writing guide, boolean methods prefer true/false wording over "whether". Try:

    If [code]true[/code] the given column is selectable.

File: doc/classes/TreeItem.xml [hunk 2]
```diff
@@ -433,6 +474,7 @@
 			<argument index="1" name="texture" type="Texture">
 			</argument>
 			<description>
+				Sets the icon [Texture] for column [code]column[/code].
```
Comment (cbscribe) [inline doc/classes/TreeItem.xml]: Avoid "of the" constructions where possible:

Sets the given column's icon [Texture].

File: doc/classes/Tree.xml [hunk 3]
```diff
@@ -33,7 +33,7 @@
 			<return type="void">
 			</return>
 			<description>
-				Clear the tree. This erases all of the items.
+				Clear the tree. This removes all items.
```
Comment (cbscribe) [inline doc/classes/Tree.xml]: Clear -> Clears

File: doc/classes/Tree.xml [hunk 4]
```diff
@@ -49,13 +49,14 @@
 			<return type="void">
 			</return>
 			<description>
-				Make the current selected item visible. This will scroll the tree to make sure the selected item is in sight.
+				Make the currently selected item visible. This will scroll the tree to make sure the selected item is in sight.
```
Comment (cbscribe) [inline doc/classes/Tree.xml]: Make -> Makes

"is in sight" -> "is visible"

**Review by cbscribe — CHANGES_REQUESTED** [review]
I didn't comment on every line, but in general:
    - use "the column's icon" instead of "the icon of the column
    - use [code]true[/code] and [code]false[/code] for boolean members/methods

**Comment by akien-mga** [comment]
Please make sure to squash the commits together when making the requested changes: http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase

File: doc/classes/TreeItem.xml [hunk 5]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="TreeItem" inherits="Object" category="Core" version="3.0.alpha.custom_build">
 	<brief_description>
+		Control for a single item inside a [Tree].
 	</brief_description>
 	<description>
+		Control for a single item inside a [Tree]. May have child [TreeItem]s and be styled as well as contain buttons.
```
Comment (Noshyaar) [inline doc/classes/TreeItem.xml]: `[TreeItem]\ s` with a backslash and a space, otherwise the link breaks.

File: doc/classes/TreeItem.xml [hunk 6]
```diff
@@ -303,6 +333,7 @@
 			<argument index="0" name="column" type="int">
 			</argument>
 			<description>
+				Selects the colum [code]column[/code].
```
Comment (Noshyaar) [inline doc/classes/TreeItem.xml]: column

**Comment by mhilbrunner** [comment]
Done & squashed. Implemented all suggestions above, hope I haven't missed anything :)

File: doc/classes/Tree.xml [hunk 7]
```diff
@@ -80,7 +81,7 @@
 			<argument index="0" name="column" type="int">
 			</argument>
 			<description>
-				Get the title of the given column.
+				Get the title column's title.
```
Comment (cbscribe) [inline doc/classes/Tree.xml]: You have an extra "title" and getters should use "Returns".

File: doc/classes/Tree.xml [hunk 8]
```diff
@@ -89,7 +90,7 @@
 			<argument index="0" name="column" type="int">
 			</argument>
 			<description>
-				Get the width of the given column in pixels.
+				Get the column's width in pixels.
```
Comment (cbscribe) [inline doc/classes/Tree.xml]: Returns

File: doc/classes/Tree.xml [hunk 9]
```diff
@@ -110,7 +111,7 @@
 			<return type="int">
 			</return>
 			<description>
-				Get the flags of the current drop mode.
+				Get the current drop mode's flags.
```
Comment (cbscribe) [inline doc/classes/Tree.xml]: Returns

File: doc/classes/Tree.xml [hunk 10]
```diff
@@ -125,14 +126,14 @@
 			<return type="TreeItem">
 			</return>
 			<description>
-				Get the current edited item. This is only available for custom cell mode.
+				Get the currently edited item. This is only available for custom cell mode.
```
Comment (cbscribe) [inline doc/classes/Tree.xml]: Returns

File: doc/classes/Tree.xml [hunk 11]
```diff
@@ -125,14 +126,14 @@
...
 			</description>
 		</method>
 		<method name="get_edited_column" qualifiers="const">
 			<return type="int">
 			</return>
 			<description>
-				Get the column of the cell for the current edited icon. This is only available for custom cell mode.
+				Get the column for the currently edited item. This is only available for custom cell mode.
```
Comment (cbscribe) [inline doc/classes/Tree.xml]: Returns

File: doc/classes/Tree.xml [hunk 12]
```diff
@@ -143,7 +144,7 @@
 			<argument index="1" name="column" type="int" default="-1">
 			</argument>
 			<description>
-				Get the rectangle area of the the specified item. If column is specified, only get the position and size of that column, otherwise get the rectangle containing all columns.
+				Get the rectangle area for the specified item. If column is specified, only get the position and size of that column, otherwise get the rectangle containing all columns.
```
Comment (cbscribe) [inline doc/classes/Tree.xml]: Returns

File: doc/classes/Tree.xml [hunk 13]
```diff
@@ -168,14 +169,14 @@
 			<return type="int">
 			</return>
 			<description>
-				Get the index of the last pressed button.
+				Get the last pressed button's index.
```
Comment (cbscribe) [inline doc/classes/Tree.xml]: Returns

File: doc/classes/Tree.xml [hunk 14]
```diff
@@ -168,14 +169,14 @@
...
 			</description>
 		</method>
 		<method name="get_root">
 			<return type="TreeItem">
 			</return>
 			<description>
-				Get the root item of the tree.
+				Get the tree's root item.
```
Comment (cbscribe) [inline doc/classes/Tree.xml]: Returns

File: doc/classes/Tree.xml [hunk 15]
```diff
@@ -196,14 +197,14 @@
 			<return type="int">
 			</return>
 			<description>
-				Get the column number of the current selection.
+				Get the current selection's column.
```
Comment (cbscribe) [inline doc/classes/Tree.xml]: Returns

File: doc/classes/Tree.xml [hunk 16]
```diff
@@ -212,6 +213,7 @@
 			<argument index="0" name="allow" type="bool">
 			</argument>
 			<description>
+				Set whether a cell that is currently already selected may be selected again.
```
Comment (cbscribe) [inline doc/classes/Tree.xml]: Don't use "set whether", it's more explicit to use [code]true[/code]. 
"If [code]true[/code] the currently selected cell may be selected again."

File: doc/classes/Tree.xml [hunk 17]
```diff
@@ -307,7 +309,7 @@
 			<argument index="0" name="mode" type="int" enum="Tree.SelectMode">
 			</argument>
 			<description>
-				Set the selection mode. Use one of the [code]SELECT_*[/code] constants.
+				Set the selection mode. See the [code]SELECT_*[/code] constants.
```
Comment (cbscribe) [inline doc/classes/Tree.xml]: Allows single or multiple selection. See...

File: doc/classes/Tree.xml [hunk 18]
```diff
@@ -298,7 +300,7 @@
 			<argument index="0" name="enable" type="bool">
 			</argument>
 			<description>
-				Set whether the root of the tree should be hidden.
+				Set whether the tree's root should be hidden.
```
Comment (cbscribe) [inline doc/classes/Tree.xml]: Boolean

File: doc/classes/TreeItem.xml [hunk 19]
```diff
@@ -413,6 +458,7 @@
 			<argument index="1" name="enabled" type="bool">
 			</argument>
 			<description>
+				If [code]true[/code], column [code]column[/code] is editable.
```
Comment (cbscribe) [inline doc/classes/TreeItem.xml]: Remove comma

File: doc/classes/TreeItem.xml [hunk 20]
```diff
@@ -423,6 +469,7 @@
 			<argument index="1" name="enable" type="bool">
 			</argument>
 			<description>
+				If [code]true[/code], column [code]column[/code] is expanded to the right.
```
Comment (cbscribe) [inline doc/classes/TreeItem.xml]: Remove comma

**Review by cbscribe — CHANGES_REQUESTED** [review]
Did a fairly quick pass. Very good - it's all minor wording changes.

**Comment by mhilbrunner** [comment]
@cbscribe: Done.

**Review by cbscribe — APPROVED** [review]

**Comment by cbscribe** [comment]
Thanks for your efforts - this was a big job.

**Pull request merged**